nni/algorithms/compression/pytorch/pruning/structured_pruning_masker.py

```diff
@@ -33,11 +33,12 @@ class StructuredWeightMasker(WeightMasker):
 
     """
 
-    def __init__(self, model, pruner, preserve_round=1, dependency_aware=False):
+    def __init__(self, model, pruner, preserve_round=1, dependency_aware=False, global_sort=False):
```

Comment:
I think it may be better to implement it for Taylor alone without changing the underlying interface, because we don't have another masker use these interfaces, and `slim` is natural using global sort.

Reply:
But based our discussion before, we agree that we should modify the condition in 

> https://github.com/microsoft/nni/blob/master/nni/algorithms/compression/pytorch/pruning/structured_pruning_masker.py#L63

May be we can add `assert` to tell user that we only support Taylor currently?

Reply:
the current implementation is good for me, just a little opinion, don't need to modify. And we need update doc for Taylor.

Reply:
Have updated doc.

Reply:
so you put global sort in `StructuredWeightMasker` but not in `TaylorFOWeightFilterPrunerMasker`. what is the reason for it? is it because this would make `TaylorFOWeightFilterPrunerMasker` have different initial arguments with other structured weight masker?

Reply:
Because in [here](https://github.com/microsoft/nni/pull/3896/files#diff-3c3764e5ee06fcf6e53f6bf6a96fe6d9c493906106f6714b040590cb2ab35255R64), we modify the conditional judgement statement which discriminate different situations including global-sort and dependency-aware. The reason why we do conditional judgement [here](https://github.com/microsoft/nni/pull/3896/files#diff-3c3764e5ee06fcf6e53f6bf6a96fe6d9c493906106f6714b040590cb2ab35255R64) is because we think global-sort should be the same level with dependency-aware. This conditional judgement is done in `StructuredWeightMasker ` so we put `global_sort` here.